src/sentry/mediators/service_hooks/destroyer.py

```diff
@@ -0,0 +1,14 @@
+from __future__ import absolute_import
+
+from sentry.mediators import Mediator, Param
+
+
+class Destroyer(Mediator):
+    service_hook = Param('sentry.models.ServiceHook')
+
+    def call(self):
+        self._destroy_service_hook()
+        return self.service_hook
```

Comment:
Does the caller need a deleted record for anything?

Reply:
I think this is mostly just to be consistent with what I did in other mediators. And I did it just because I'm used to returning a representation of the object, even in `DELETE` endpoints, tbh.

I think the logic behind that was basically this part of the RFC:

> a 200 (OK) status code if the action has been enacted and the response message includes a representation describing the status.

coupled with the idea that if an object isn't immediately deleted, you'd return a `202 Accepted` with a representation of the object, since it technically still exists.

So to be consistent between those two scenarios, it made sense to always return a representation.

With that said, I'm not really sure when this would practically be necessary, so 🤷‍♂️

Reply:
`DELETE` can also use `204` if the object is gone immediately.

Reply:
Yea. I think the endpoint can choose what to expose.

I suppose there'd be some value in these objects interfaces' conforming to those representations (204 No Content == None return value, etc.), but then again there's probably also value in all mediators consistently returning domain objects. Especially when they're meant to be composed in interesting ways, wrapped in a single transaction.

I don't have overly strong opinions here, other than I don't think returning the object is overly harmful or surprising/confusing and leaves the door open for instances where it might be useful.